src/Controller/CRUDController.php

```diff
@@ -1555,6 +1555,38 @@ protected function preList(Request $request)
         return $this->get('translator')->trans($id, $parameters, $domain, $locale);
     }
 
+    protected function handleXmlHttpRequestErrorResponse(Request $request, FormInterface $form): ?JsonResponse
+    {
+        if (empty(array_intersect(['application/json', '*/*'], $request->getAcceptableContentTypes()))) {
+            @trigger_error(sprintf('"%s" is not supported since sonata-project/admin-bundle 3.x and the status code 406 will be set in the returned response in 4.0. Use `application/json`.', implode('", "', $request->getAcceptableContentTypes())), E_USER_DEPRECATED);
```

Comment:
This is just a suggestion. I also think that this message could be improved even more in cases where the "Accept" header is omitted, in order to avoid showing `""` as the passed value.

```suggestion
            @trigger_error(sprintf(
                'None of the passed values ("%s") in the "Accept" header when requesting %s %s is supported since sonata-project/admin-bundle 3.x.'
                .' It will result in a response with the status code 406 (Not Acceptable) in 4.0. You must add `application/json`.',
                implode('", "', $request->getAcceptableContentTypes()),
                $request->getMethod(),
                $request->getUri()
            ), E_USER_DEPRECATED);
```